### PR TITLE
updated scala3-book ch1 and ch2 according to latest English Version

### PR DESCRIPTION
**Pull request by benluo**

### CODE REVIEW

File: _zh-cn/overviews/scala3-book/scala-features.md [hunk 1]
```diff
@@ -122,8 +117,39 @@ def double(ints: List[Int]): List[Int] = {
   buffer.toList
 }
 
+val newNumbers = double(oldNumbers)
+```{% tabs scala-features-1 class=tabs-scala-version %}
```
Comment (bishabosha) [inline _zh-cn/overviews/scala3-book/scala-features.md:121]:
```suggestion
{% tabs scala-features-1 class=tabs-scala-version %}
```

File: _zh-cn/overviews/scala3-book/introduction.md [hunk 2]
```diff
@@ -5,26 +5,25 @@ description: This page begins the overview documentation of the Scala 3 language
 num: 1
 previous-page: 
 next-page: scala-features
-
-scala3: true
-partof: scala3-book
-overview-name: "Scala 3 — Book"
-layout: multipage-overview
-permalink: "/zh-cn/scala3/book/:title.html"
```
Comment (bishabosha) [inline _zh-cn/overviews/scala3-book/introduction.md:13]: This permalink needs to be applied across the whole translated book for it to render correctly

**Comment by benluo** [comment]
I don't know why the compiler always show that 

```
Checking 1483 external links...
- ./_site/zh-cn/scala3/book/introduction.html
Ran on 1[17](https://github.com/scala/docs.scala-lang/actions/runs/3109232296/jobs/5039269005#step:7:18)5 files!
  *  internally linking to scala-features.html, which does not exist (line 554)
     <a href="scala-features.html"><strong>next</strong> →</a>

HTML-Proofer found 1 failure!
```

File: _zh-cn/overviews/scala3-book/scala-features.md [hunk 3]
```diff
@@ -4,15 +4,10 @@ type: chapter
 description: This page discusses the main features of the Scala 3 programming language.
 num: 2
 previous-page: introduction
-next-page: 
-
-scala3: true
-partof: scala3-book
-overview-name: "Scala 3 — Book"
-layout: multipage-overview
-permalink: "/zh-cn/scala3/book/:title.html"
```
Comment (bishabosha) [inline _zh-cn/overviews/scala3-book/scala-features.md:13]: again, it seems lines 9-13 is necessary across each page in the book

File: _zh-cn/overviews/scala3-book/introduction.md (hunk 2 above)
Comment (benluo) [reply _zh-cn/overviews/scala3-book/introduction.md]: add.

File: _zh-cn/overviews/scala3-book/scala-features.md (hunk 3 above)
Comment (benluo) [reply _zh-cn/overviews/scala3-book/scala-features.md:13]: add.

**Comment by benluo** [comment]
i will correct them and pr separately.

**Pull request closed without merging**